packages/financial-templates-lib/src/price-feed/USPACPriceFeed.ts

```diff
@@ -0,0 +1,251 @@
+    private readonly minTimeBetweenUpdates = 3600 // 1 hour is enough to fit in free plan (1000 requests per day)
+  ) {
+    super();
+
+    assert(symbols != null, "symbols must be provided");
+    assert(rapidApiKey != null, "rapidApiKey must be provided");
+    assert(correctionFactor != null, "correctionFactor must be provided");
+
+    this.uuid = `USPACPriceFeed-${symbols.join(",")}`;
+  }
+
+  public getCurrentPrice(): BN | null {
+    return this.currentPrice;
+  }
+
+  public async getHistoricalPrice(time: number, ancillaryData: string, verbose = false): Promise<BN> {
+    // Construct URL.
+    // https://rapidapi.com/principalapis/api/stock-data-yahoo-finance-alternative/
+    const symbolsStr = this.symbols.join("%2C");
+    const url = `https://stock-data-yahoo-finance-alternative.p.rapidapi.com/v8/finance/spark?symbols=${symbolsStr}&range=3mo&interval=1d`;
+    const options = {
+      method: "GET",
+      headers: {
+        "x-rapidapi-host": "stock-data-yahoo-finance-alternative.p.rapidapi.com",
+        "x-rapidapi-key": this.rapidApiKey,
+      },
+    };
+    const response = (await this.networker.getJson(url, options)) as RapidAPIHistoryReponse | null | undefined;
+
+    if (response == null) {
+      throw new Error(`🚨Invalid response from url ${url}`);
+    }
+
+    if (verbose) {
+      console.log(`Rapid API response for url ${url} is ${JSON.stringify(response, null, 2)}`);
+    }
+
+    // Validate response
+    for (const symbol of this.symbols) {
+      const stockData = response[symbol];
+      if (stockData == null) {
+        throw new Error(`🚨Could not parse price result from url ${url}: missing data for symbol ${symbol}`);
+      }
+      if (stockData.timestamp.length == 0 || stockData.timestamp.length != stockData.close.length) {
+        throw new Error(`🚨Could not parse price result from url ${url}: invalid data for symbol ${symbol}`);
+      }
+      for (const ts of stockData.timestamp) {
+        if (typeof ts != "number") {
+          throw new Error("Could not parse data: invalid timestamp type");
+        }
+      }
+      for (const p of stockData.close) {
+        if (typeof p != "number") {
+          throw new Error("Could not parse data: invalid price type");
+        }
+      }
+      for (let i = 1; i < stockData.timestamp.length; i++) {
+        if (stockData.timestamp[i] <= stockData.timestamp[i - 1]) {
+          throw new Error(`Could not parse data: timestamps for ${symbol} are not increasing monotonously`);
+        }
+      }
+    }
+
+    // Get prices
+    const prices = [];
+    for (const symbol of this.symbols) {
+      const stockData = response[symbol];
+      /*
+        if time is greater than last timestamp (that is equal to the beginning
+        of the last market open datetime) plus 24hours then new day begin and
+        we should use current price
+      */
+      if (time >= stockData.timestamp[stockData.timestamp.length - 1] + 3600 * 24) {
```

Comment:
This logic is not specified in UMIP and would result in different price for the same request timestamp just depending on when one runs this script.

Reply:
Reworked to use intraday price